src/devices/sound/qsoundhle.cpp

```diff
@@ -82,74 +68,120 @@ void qsound_hle_device::device_start()
 {
 	m_stream = stream_alloc(0, 2, clock() / 2 / 1248); // DSP program uses 1248 machine cycles per iteration
 
-	// create pan table
-	for (int i = 0; i < 33; i++)
-		m_pan_table[i] = (int)((256 / sqrt(32.0)) * sqrt((double)i));
+	init_register_map();
 
-	// init sound regs
-	memset(m_channel, 0, sizeof(m_channel));
+	// state save
+	// PCM registers
+	for(int j=0;j<16;j++) // PCM voices
```

Comment:
Please follow the convention used in existing files - space between control keyword and opening parenthesis of control expression, and spaces after semicolon/comma separators.